docs: Add OpenBanking API documentation and OpenAPI specification

kshitijk4poor opened the pull request:
- Added comprehensive OpenBanking API documentation with detailed endpoints
- Created OpenAPI specification for OpenBanking purchases and status retrieval
- Included MDX documentation for:
  * OpenBanking API overview
  * Get OpenBanking Status endpoint
  * Webhooks configuration and handling
- Updated mint.json to include OpenBanking section in navigation
- Added detailed examples, use cases, and best practices for OpenBanking payments

kshitijk4poor committed:
docs: Add OpenBanking API documentation and OpenAPI specification

- Added comprehensive OpenBanking API documentation with detailed endpoints
- Created OpenAPI specification for OpenBanking purchases and status retrieval
- Included MDX documentation for:
  * OpenBanking API overview
  * Get OpenBanking Status endpoint
  * Webhooks configuration and handling
- Updated mint.json to include OpenBanking section in navigation
- Added detailed examples, use cases, and best practices for OpenBanking payments

kshitijk4poor committed:
removed steps

unnati914 commented on mint.json:
```diff
@@ -132,7 +138,15 @@
+          "pages": [
+            "api-reference/openbanking/openbanking-api",
+            "api-reference/openbanking/get-status",
+            "api-reference/openbanking/webhooks"
```
shouldn't be another group ideally

unnati914 commented on mint.json:
```diff
@@ -80,6 +80,12 @@
             "pages/payment-methods/pix/pix-pay-in-out"
           ]
         },
+        {
```
you can remove this page

kshitijk4poor committed:
docs: Remove OpenBanking API from navigation and documentation

- Removed OpenBanking API section from mint.json navigation
- Removed OpenBanking API reference pages from API Reference section
- Kept OpenBanking API cards in introduction.mdx for reference
